rulebook.tex

```diff
@@ -1008,23 +1019,38 @@ \subsubsection{MPS --- During Production Phase}
 fed into the machine. It receives a ring of the desired color and
 moves the processed product to the output.
 
-\noindent\textbf{SS}\footnote{The TC is working towards the
-  integration of the SS for 2019. Issues arising or the lack of
-  development resources might lead to exclusion of the SS from the
-  tournament on short notice.}
-%
-The SS prepare message denotes the storage place where to store the
-next input workpiece, or from which place to retrieve an item
-from. There will be designated positions for specific $C_0$ workpieces
-which will be announced at the beginning of the
-tournament\footnote{Please note again, this is an assignment
-  specifically for 2019 and may change in future competitions.}. If a
-position is requested for which no product has been stored (or has
-already been retrieved), the machine will be broken
-(cf.~\refsec{sec:broken-machine}). The products retrieved from the SS
-are only allowed for some specific order,
-cf.~\refsec{sec:production-plan}. Note the costs for retrieving a base
-outlined in \refsec{sec:production-scoring}.
+\noindent\textbf{SS}
+The SS prepare message denotes the storage place (shelf and slot number) and
+the type of the operation that should be performed.
+The SS consists of six shelves with eight slots each. \reffig{fig:ss-shelf}
+depicts how positions are mapped to their respective shelf and slot numbers
+$(x,y),x\in[0,5],y\in[0,7]$.
+Accessing a slot $(x,y), y\in\{1,3,5,7\}$ in the back of a shelf while slot
+$(x,y-1)$ is occupied causes the SS to relocate the product at $(x,y-1)$ to a
+randomly chosen accessible position before processing the instruction.
+If all other free positions are located in the back and
+are also blocked by products stored in front of them, then one of those free
+positions $(x^\prime,y^\prime)$ is chosen and
+$(x^\prime,y^\prime-1)$ is relocated to $(x^\prime,y^\prime)$, before
+relocating $(x,y-1)$ to $(x^\prime,y^\prime)$.
+Descriptions can be added to the prepare messages when storing workpieces.
+These descriptions are attached to the position of the workpiece and updated
+automatically when relocating operations happen.
+The storage status of each position is periodically sent to each team,
+including the corresponding descriptions.
```

Comment:
I assume each team only gets the info about their own SS? This is not entirely clear here

Reply:
> I assume each team only gets the info about their own SS? This is not entirely clear here

True, i fixed it in https://github.com/robocup-logistics/rcll-rulebook/pull/40/commits/0f81e123205e0977e15fcd41aa24193fe1e708b2